src/test.cpp

```diff
@@ -109,14 +109,15 @@ void TestExhaustive(uint32_t bits, size_t capacity) {
                 // Compare
                 CHECK(serialized == serialized_rebuild);
                 // Count it
-                if (elements_0.size() <= capacity) ++counts[elements_0.size()];
+                if (impl == 0 && elements_0.size() <= capacity) ++counts[elements_0.size()];
             }
         }
     }
 
     // Verify that the number of decodable sketches with given elements is expected.
-    for (uint64_t i = 0; i <= capacity && i >> bits; ++i) {
-        CHECK(counts[i] == Combination((uint64_t{1} << bits) - 1, i));
+    uint64_t mask = bits == 64 ? UINT64_MAX : (uint64_t{1} << bits) - 1;
```

Comment:
And this is the real change: replacing `i >> bits` with `(uint64_t{1} << bits) - 1`?

Reply:
There were 3 issues:
* The comparison between `counts` and `Combination()` wasn't running at all ((i >> bits) is always 0 for i==0, so the loop would instantly exit.
* The `counts` values were wrong (because it was counting a sum across all implementation), so *if* they would be compared to something, the test would incorrectly fail.
* `(i >> bits)` is undefined when bits==64.

The first would be fixed by changing `i >> bits` to `(i >> bits) == 0`, but that leaves the `>> 64` issue, so I introduce this mask and use a different way of writing the same, that keeps working for bits==64.
The second is fixed by only incrementing `counts[...]` for impl==0.

Reply:
Updated the PR description with this explanation.